### ISSUE TITLE
Checking "packages.linuxmint.com" instead of the user's source settings

### ISSUE THREAD

**Issue by Com2486** [linuxmint/mintupgrade#116]
I can not access "packages.linuxmint.com", so I changed it to a local source.
Despite this, "usr/lib/checks.py" checks for "packages.linuxmint.com" instead of the user's source settings.
Since I can not access that site, I am stuck on 21.3 until this is fixed.

**Comment by CJE46** [comment]
Edit the checks.py file and replace all references to "packages.linuxmint.com" with another repo source such as "repo.c48.uk/mint/pkgs" and save the file and run mintupgrade again. Worked for me.

**Comment by Com2486** [comment]
That works fine, but shouldn't the tools always check the users sources?

**Comment by Jeremy7701** [comment]
I can access "packages.linuxmint.com" - but only via http - not https. It appears to contain all mint packages for a number of past and current releases.

**Comment by drgrumpy** [comment]
Also affects me on mint 21.3 xfce.... as above it seems packages.linuxmint.com is hardcoded, and is broken, not accessible for me at present in the UK... but are present in mirror servers, I replaced with https://mirror.server.net/linuxmint/packages/

mintupgrade then proceeds

**Comment by InfoLibre** [comment]
packages.linuxmint.com is no longer working for some users: https://forums.linuxmint.com/viewtopic.php?p=2683237. Could you please keep the local sources during an upgrade instead of automatically replacing them with packages.linuxmint.com?

**Comment by InfoLibre** [comment]
I can't find the checks.py file. find / -name checks.py doesn't return any results. Where can I manually change upgrade sources?

**Comment by Jerome-Herbinet** [comment]
Same awkward situation for me; impossible to upgrade from LM 22.1 to 22.2 because of http://packages.linuxmint.com/ and the fact that the update system ignores the local repository I chose and insists on using http://packages.linuxmint.com/.

**Comment by InfoLibre** [comment]
If a developer needs to test this, I can give an access to my laptop with https://www.dwservice.net.

**Comment by InfoLibre** [comment]
mintinstall updated today but now, I can not anymore change my repositories. Window to change repositories never opens.

**Comment by SaintBol** [comment]
I investigated the matter after reading a conversation (in french unfortunately, as I didn't expect to post here): https://lafibre.info/tutoriels-linux/orangesosh-bloque-les-mises-a-niveau-et-les-mises-a-jour-de-linux-mint/msg1134325/#msg1134325

My conclusion is that there might be an MTU issue on a router of the hoster of the server (TZulo), at their interconnection with their transit provider PCCW/Console, (4 bytes too small somewhere, big [1500 bytes] packets to the clients getting dropped).

**Comment by Angedestenebres** [comment]
@SaintBol Thanks for this link. I'll add it to the Orange thread: https://communaute.orange.fr/t5/espace-de-convivialit%C3%A9/Probl%C3%A8me-de-mise-%C3%A0-jour-Linux-Mint/m-p/3339515

@clefebvre @mtwebster : Hey 🐱 

I know the issue with ISP or network compenant is not on your side. However, the original subject is about upgrades (minor or major) and the script that doesn't take into account the local miror.
Maybe you have some reason to do that and your explanations would be welcome.

Moreover, when there is an issue to access to packages.linuxmint.com (no matter the reason), users are currently blocks to proceed with upgrades even if they use local miror and from my point of view it should be changed.

Thanks in advance for your feedback & sorry for the ping!

**Comment by InfoLibre** [comment]
It seems that Tuzlo has a problem, its website has been inaccessible for a week too.
https://lafibre.info/tutoriels-linux/orangesosh-bloque-les-mises-a-niveau-et-les-mises-a-jour-de-linux-mint/new/
If it is a Linux Mint provider, can somebody contact them?

**Comment by SaintBol** [comment]
True, same MTU and dropped packets issue, I documented again this one on the link you mentioned (Answer #10) ; in French, but tcpdump captures are not in a specific language :)

**Comment by patmenier** [comment]
> [@SaintBol](https://github.com/SaintBol) Thanks for this link. I'll add it to the Orange thread: https://communaute.orange.fr/t5/espace-de-convivialit%C3%A9/Probl%C3%A8me-de-mise-%C3%A0-jour-Linux-Mint/m-p/3339515
> 
> [@clefebvre](https://github.com/clefebvre) [@mtwebster](https://github.com/mtwebster) : Hey 🐱
> 
> I know the issue with ISP or network compenant is not on your side. However, the original subject is about upgrades (minor or major) and the script that doesn't take into account the local miror. Maybe you have some reason to do that and your explanations would be welcome.
> 
> Moreover, when there is an issue to access to packages.linuxmint.com (no matter the reason), users are currently blocks to proceed with upgrades even if they use local miror and from my point of view it should be changed.
> 
> Thanks in advance for your feedback & sorry for the ping!

Perhaps micro$oft has payed the ISP to block connections . Remember it's the last day of free upgrade
